.pre-commit-config.yaml

```diff
@@ -0,0 +1,24 @@
+# Exclude some file types from automatic code style
+exclude: \.(json|csv)$
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.6.0
+  hooks:
+    - id: check-merge-conflict
+    - id: check-toml
+    - id: check-yaml
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
+- repo: https://github.com/PyCQA/isort
+  rev: 5.13.2
+  hooks:
+    - id: isort
+      name: isort
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.6.8
+  hooks:
+    # Run the linter and fix simple issues automatically
+    - id: ruff
```

Comment:
Did a developer needs `ruff` installed? If so, this should be also added as a dependency.

Reply:
No, only `pre-commit`. I can open a follow-up PR to summarize this in a `CONTRIBUTING.md`.

Usually the development dependencies go into something like a `requirements-dev.txt` because they are not needed at runtime. (I can make a PR for that too.)

Reply:
Yes, dev requirements sound good.

Reply:
I would suggest switching to poetry this would enable using one single pyproject.toml for dependencies and other important configuration like versioning or e.g. ruff settings

Reply:
pyproject.toml as the single point for configuration and package metadata works well without poetry!
After reducing the number of dependencies I don't see why one would need to introduce an advanced dependency management tool that is rarely used by beginners.

Reply:
Or without poetry.